source/blood/src/map2d.cpp

```diff
@@ -216,7 +216,7 @@ void CViewMap::Draw(void)
 
 void CViewMap::Process(int nX, int nY, short nAng)
 {
-    nZoom = gZoom;
+    nZoom = gZoom*((ydim/200)>>1); // scale to display resolution
```

Comment:
```suggestion
    nZoom = scale(gZoom, ydim, 200) >> 1; // scale to display resolution
```
I believe this gives the intended result while preserving precision and avoiding overflow.